src/main/java/com/purbon/search/fair/utils/AlphaAdjuster.java

```diff
@@ -87,7 +97,7 @@ public double computeSuccessProbability() {
         while (minProtected < maxProtected) {
             //get the current blockLength from auxMTable
             int blockLength = auxMTable.at(minProtected, "block");
-            if (blockLength <= pmfCache.size() && pmfCache.get(blockLength) != null) {
+            if (blockLength < pmfCache.size() && pmfCache.get(blockLength) != null) {
```

Comment:
as a general thought, you need to avoid comparing with null.

Reply:
yes but in this case i either have to add "0" values or something to the pmfCache or do it like that to have the same behavior as in python : /

Reply:
for now it works, but as a general case, is better to just compare with empty.